src/core/authentication/authentication.service.ts

```diff
@@ -41,6 +41,7 @@ export class AuthenticationService {
   public async getAgentInfo(opts: {
```

Comment:
can you please try to simplify `getAgentInfo` it has optional fields and many branches.
I suggest to ignore the catch clause